src/main/java/org/springframework/data/solr/core/SolrTemplate.java

```diff
@@ -89,8 +90,9 @@
  * @author Joachim Uhrlass
  * @author Francisco Spaeth
  * @author Shiradwade Sateesh Krishna
+ * @author Venil Noronha
  */
-public class SolrTemplate implements SolrOperations, InitializingBean, ApplicationContextAware {
+public class SolrTemplate implements SolrOperations, MulticoreSolrOperations, InitializingBean, ApplicationContextAware {

```

Comment:
Thank you @rene-d-menoto for your inputs! I'll give some more thought to find if there is a better way.